client/components/calypso-i18n-provider/index.tsx

```diff
@@ -2,21 +2,23 @@
  * External dependencies
  */
 import * as React from 'react';
-import { setLocaleData as setWpI18nLocaleData } from '@wordpress/i18n';
-import { I18nProvider } from '@automattic/react-i18n';
+import { createI18n, setLocaleData as setWpI18nLocaleData } from '@wordpress/i18n';
+import { I18nProvider } from '@wordpress/react-i18n';
 import i18n from 'i18n-calypso';
 import { LocaleProvider, i18nDefaultLocaleSlug } from '@automattic/i18n-utils';
 
 const CalypsoI18nProvider: React.FunctionComponent = ( { children } ) => {
 	const [ localeData, setLocaleData ] = React.useState( i18n.getLocale() );
 	const [ localeSlug, setLocaleSlug ] = React.useState( i18n.getLocaleSlug() );
+	const wpI18n = React.useMemo( () => createI18n( localeData ), [ localeSlug ] );
```

Comment:
Given we're already calling `wpI18n.setLocaleData()` when the locale changes, do we really need to recreate this object when the slug changes?

Using `useRef()` would be a way of having a single I18n instance for the lifetime of the component.

```suggestion
	const wpI18n = React.useRef( createI18n( localeData ) );
```

Reply:
That's what I was thinking too.

And even if we kept the `useMemo` call, shouldn't the dependency be on `localeData` (instead of `localeSug`?)